### PR TITLE
Tweak incorrect escaped char diagnostic

### PR DESCRIPTION
**Pull request by estebank**

### CODE REVIEW

**Review by davidtwco — APPROVED** [review]

File: src/test/ui/parser/byte-string-literals.stderr [hunk 1]
```diff
@@ -2,13 +2,13 @@ error: unknown byte escape: f
   --> $DIR/byte-string-literals.rs:6:32
    |
 LL | static FOO: &'static [u8] = b"/f";
-   |                                ^
+   |                                ^ unknown byte escape
 
 error: unknown byte escape: f
   --> $DIR/byte-string-literals.rs:9:8
    |
 LL |     b"/f";
```
Comment (eddyb) [inline src/test/ui/parser/byte-string-literals.stderr]: Do we really normalize backslashes to slashes?! Can we avoid it on lines with `'` or `"` on them? cc @Mark-Simulacrum @oli-obk @nikomatsakis

Comment (estebank) [reply src/test/ui/parser/byte-string-literals.stderr]: We do for stderr files with no heuristic to avoid the current output.

**Pull request merged**